### DISCUSSION TITLE
Do not raise an error if the language of the file extension cannot be determined

### DISCUSSION THREAD

**Discussion by smorimoto** [biomejs/biome#1479] [General]
As the title says, it currently raises an error if the language of the file extension cannot be determined, but it should be either warning or info.

**Comment by smorimoto** [comment]
Although the log shows it as warning, it somehow ended up being reported as error, and the exit code is also set as error 🤷‍♂️

```
xxxx files/missingHandler ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ⚠ Biome could not determine the language for the file extension ml
  

The number of diagnostics exceeds the number allowed by Biome.
Diagnostics not shown: 161.
Compared 81 file(s) in 11ms
format ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Some errors were emitted while running checks.
```

```
❯ echo $status
1
```

**Reply by ematipico** [reply]
> Diagnostics not shown: 161.

There are 161 diagnostics. Are you sure that all 161 aren't errors?

**Reply by smorimoto** [reply]
Perhaps the lack of support for nested ignore files caused too many errors, most of which were not printed in the log, causing confusion.

**Comment by Conaclos** [comment]
You can set [`files.ignoreUnknown`](https://biomejs.dev/reference/configuration/#filesignoreunknown) to true or the CLI option `files-ignore-unknown=true` top ignore unknown extensions.

**Reply by smorimoto** [reply]
Warning seems correct behaviour, but is getting an error correct?

**Reply by Conaclos** [reply]
No I think there is an issue there. I will turn the discussion into an issue.

**Reply by smorimoto** [reply]
Thanks a lot!

**Reply by EvHaus** [reply]
Is there a reason `"ignoreUnknown": true` isn't the default? Seems weird to me for Biome to be (by default) complaining about files it can't parse.

**Reply by ematipico** [reply]
> Is there a reason `"ignoreUnknown": true` isn't the default? Seems weird to me for Biome to be (by default) complaining about files it can't parse.

This was the default behaviour since the beginning.

Since Biome (formerly Rome), was designed to be a bundler and handle many kinds of files, it was supposed to warn users for extraneous files.

Happy to review the behaviour.

**Comment by Conaclos** [comment]
This should now be solved :)
